## Урок 2. Домашнее задание. 
## Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

In [1]:
import numpy as np

### Задача 2.1

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите $M(X)$, $D(X)$.

#### Решение:

Возможные значения случайной величины (в нашем случае правильных ответов) при простом угадывании при пяти вопросах: $x_1 = 0$, $x_2 = 1$, $x_3 = 2$, $x_4 = 3$, $x_5 = 4$, $x_6 = 5$. В задании пять вопросов, а значит мы имееем 5 независимых испытаний. Вероятность правильно угадать ответ из четырех вопросов равна $p = 0.25$.

Посчитаем вероятность каждого значения:


$P(X = 0) = C_n^k p^k (1 - p)^{n - k} = C_5^0 \cdot 0.25^0 \cdot 0.75^{5-0} = \frac{5!}{0! \: (5-0)!} \cdot 0.25^0 \cdot 0.75^{5-0} = 1 \cdot 1 \cdot 0.2373 = 0.2373.$

$P(X = 1) = C_5^1 \cdot 0.25^1 \cdot 0.75^4 = \frac{5!}{1! \: 4!} \cdot 0.25^1 \cdot 0.75^4 = 5 \cdot 0.25 \cdot 0.3164 = 0.3955.$

$P(X = 2) = C_5^2 \cdot 0.25^2 \cdot 0.75^3 = 0.2636.$

$P(X = 3) = C_5^3 \cdot 0.25^3 \cdot 0.75^2 = 0.0878.$

$P(X = 4) = C_5^4 \cdot 0.25^4 \cdot 0.75^1 = 0.0146.$

$P(X = 5) = C_5^5 \cdot 0.25^5 \cdot 0.75^0 = 0.0009.$

Итак, закон распределения такой случайной величины:

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X = x)$</th>
        <td>0.2373</td>
        <td>0.3955</td>
        <td>0.2636</td>
        <td>0.0878</td>
        <td>0.0146</td>
        <td>0.0009</td>
    </tr>
</table>

Посчитаем математическое ожидание распределения правильных ответов. Поскольку $n = 5$ и $p = 0.25$, получаем:

$$M(X) = n \cdot p = 5 \cdot 0.25 = 1.25.$$

Результат можно понимать так: в среднем при пяти билетах, в к/х четыре ответа и только один правильный, ожидается  $1.25$ правильных ответа, если выбирать наугад.

Дисперсия такой величины:

$$D(X) = n \cdot p \cdot (1 - p) = 5 \cdot 0.25 \cdot (1 - 0.25) = 0.9375.$$

### Задача 2.2

Вероятность попадания в цель при одном выстреле равна $0.01$. Какова вероятность того, что число попаданий при $200$ выстрелах будет не менее $5$ и не более $10$?

#### Решение:

Вероятность того, что число попаданий будет не менее пяти и не более десяти находится в интервале соответствующих значений попаданий в мишень $x=5$ и $x=10$, при $n = 200$ и $p = 0.01$. 

Число испытаний достаточно велико, а вероятность достаточно мала, поэтому воспользуемся формулой Пуассона для нахождения этих вероятностей. Здесь $a = np = 200 \cdot 0.01 = 2$, $k_5 = 5$ и $k_{10} = 10$.

$P(X = 5) = \dfrac{a^{k} \cdot e^{-a}}{k!} = \dfrac{2^{5} \cdot e^{-2}}{5!} \approx 0.036089.$

$P(X = 5) = \dfrac{a^{k} \cdot e^{-a}}{k!} = \dfrac{2^{10} \cdot e^{-2}}{10!} \approx 0.000038.$


Проверка по формуле Бернулли:

$P(X = 5) = C_n^k p^k (1 - p)^{n - k} = C_{200}^5 \cdot 0.01^5 \cdot 0.99^{200-5} = 0,035723.$

$P(X = 10) = C_n^k p^k (1 - p)^{n - k} = C_{200}^{10} \cdot 0.01^{10} \cdot 0.99^{200-10} = 0,000033.$


Вычисления по формуле Пуассона:

In [39]:
def puasson(a, k):
    return (a ** k) * (np.exp(-a)) / np.math.factorial(k)

In [40]:
print(puasson(2, 5))
print(puasson(2, 10))

0.03608940886309672
3.8189850648779595e-05


Проверка по формуле Бернулли:

In [41]:
def bernulli(n, k, p):
    return np.math.factorial(n)/(np.math.factorial(k) * np.math.factorial(n - k)) * p**k * (1 - p)**(n - k)

In [44]:
print(bernulli(200, 5, 0.01))
print(bernulli(200, 10, 0.01))

0.03572335672160829
3.326003843593443e-05


Таким образом вероятость того, что будет не менее 5 и не более 10 попаданий находится в интервале $[0.000033, 0.035723]$.

С учетом того, что математическое ожидание $M(X) = a = n \cdot p = 200 \cdot 0.01 = 2$, вполне правдоподобный интервал вероятностей.

### Задача 2.3

В условиях предыдущей задачи сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью $0.9$, что хотя бы $10$ раз будет совершено попадание?

_Подсказка_. Как посчитать $P(X \geq k)$ для величины $X$, имеющей распределение Пуассона? Здесь предстоит немножко покодить.